piptools/writer.py

```diff
@@ -189,17 +189,18 @@ def _iter_lines(
                 yield MESSAGE_UNSAFE_PACKAGES
 
             for ireq in unsafe_requirements:
-                req = self._format_requirement(
-                    ireq,
-                    reverse_dependencies,
-                    primary_packages,
-                    marker=markers.get(key_from_ireq(ireq)),
-                    hashes=hashes,
-                )
+                ireq_key = key_from_ireq(ireq)
```

Comment:
could as well generate this above as
```python
unsafe_requirements_map = ((key_from_ireq(r), r) for r in unsafe_requirements)
```
and iterate over that

Reply:
But we still need an `ireq` (which uses in `self._format_requirement`) reference in the for-loop body.

Reply:
yes, by using `for ireq_key, ireq in unsafe_requirements_map:`

Reply:
Oh, that's tricky. I'd prefer it keep simple :)

Reply:
ok